### PR TITLE
Add PDB stuff to the Patroni upgrade doc.

### PR DESCRIPTION
**Pull request by caggles**

### CODE REVIEW

**Comment by StevenBarre** [comment]
Patroni can run with only one pod healthy? I would have thought a maxUnavailable of 1 would be better? Ie if the first node drain takes out one of the three pods, further disruptions are prevented until that one is healthy again. So if its got a disk issue, or fails to re-join properly then we don't take out a second. Doesn't quorum need 2 of 3 healthy to know which set of data is the correct one?

Def happy to have a minAvailable 1 though, as that is more likely to not block upgrades. Just want to be sure we're giving the most resilient / available advice.

**Comment by wmhutchison** [comment]
For however Patroni is deployed, does it configure the DB pods so that they do not attempt to provision on the same node? If not, then perhaps it should be considered, not as a blocker for the OCP upgrade, but just overall general improved redundancy since it is possible for a node to fail for other reasons (hardware, squirrels, etc).

https://docs.openshift.com/container-platform/4.8/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-anti-affinity-configuring_nodes-scheduler-pod-affinity

For existing nodes that might have multiple DB pods on the same node, I imagine PDB if configured would protect us from draining both pods,.

**Comment by caggles** [comment]
@StevenBarre Patroni can function with only one member, yes. But I admit I was assuming that this would only be for a short period of time, and hadn't really considered the possibility of both members being down long-term because of PVC issues or something. Having minAvailable=1 is certainly better than no PDB, which was more the perspective I was using with this. I'm open to the idea of setting minAvailable=2 or something similar, however.

@wmhutchison I don't think the Patroni pods include any specific configuration either way, but I believe that the default in openshift is for pods to have anti-affinity, so they shouldn't be on the same node if avoidable.

**Comment by StevenBarre** [comment]
Oooo how about `minAvailabe: 51%` ? That way the cluster should have quorum, regardless of the size. That will work out to 2 in a standard 3 node cluster.

The default scheduler does attempt to keep pods away from each other, but can be overridden by other priorities. I do think it is better to be explicit and set an anti-affinity policy on the pods as well to ensure they stay separated.

**Comment by caggles** [comment]
Yeah, the 51% idea sounds great, I'll add that! And add a bit about explicitly setting anti-affinity :)

File: docs/HowTo/UpdatePatroni.md [hunk 1]
```diff
@@ -2,12 +2,20 @@
...
+
+The Platform Team also advises that you explicitly set your pod anti-affinity rules for your database. OpenShift defaults to ensuring that pods start up on different nodes wherever possible, as this is the most resilient set-up. However, some considerations may supercede this requirement when left as a default. Therefore, to ensure that pods run on different nodes as often as truly possible, you should consider explicitly setting an anti-affinity configuation in your deployment.
```
Comment (wmhutchison) [inline docs/HowTo/UpdatePatroni.md:17]:
```suggestion
The Platform Team also advises that you explicitly set your pod anti-affinity rules for your database. OpenShift defaults to ensuring that pods start up on different nodes wherever possible, as this is the most resilient set-up. However, some considerations may supercede this requirement when left as a default. Therefore, to ensure that pods run on different nodes as often as truly possible, you should consider explicitly setting an [anti-affinity configuration](https://docs.openshift.com/container-platform/4.8/nodes/scheduling/nodes-scheduler-pod-affinity.html#nodes-scheduler-pod-affinity-example-antiaffinity_nodes-scheduler-pod-affinity) in your deployment.
```

**Review by StevenBarre — APPROVED** [review]

**Review by wmhutchison — APPROVED** [review]
👍

**Pull request merged**